lib/dotcom/utils/service_date_time.ex

```diff
@@ -0,0 +1,228 @@
+    |> coerce_ambiguous_time()
+    |> Timex.shift(days: 1, hours: @service_rollover_time.hour, microseconds: -1)
+    |> coerce_ambiguous_time()
+    |> Map.put(:hour, @service_rollover_time.hour - 1)
+  end
+
+  @doc """
+  Get a service range for the day of the given date_time.
+  """
+  @spec service_range_day() :: Utils.DateTime.time_range()
+  @spec service_range_day(DateTime.t()) :: Utils.DateTime.time_range()
+  def service_range_day(date_time \\ now()) do
+    beginning_of_service_day = beginning_of_service_day(date_time)
+    end_of_service_day = end_of_service_day(date_time)
+
+    {beginning_of_service_day, end_of_service_day}
+  end
+
+  @doc """
+  Get a service range for the week of the given date_time.
+  """
+  @spec service_range_current_week() :: Utils.DateTime.time_range()
+  @spec service_range_current_week(DateTime.t()) :: Utils.DateTime.time_range()
+  def service_range_current_week(date_time \\ now()) do
+    beginning_of_current_week =
+      date_time
+      |> Timex.beginning_of_week()
+      |> Timex.shift(hours: 24)
+      |> beginning_of_service_day()
+
+    end_of_current_week = date_time |> Timex.end_of_week() |> end_of_service_day()
+
+    {beginning_of_current_week, end_of_current_week}
+  end
+
+  @doc """
+  Get a service range for the week following the current week of the given date_time.
+  """
+  @spec service_range_following_week() :: Utils.DateTime.time_range()
+  @spec service_range_following_week(DateTime.t()) :: Utils.DateTime.time_range()
+  def service_range_following_week(date_time \\ now()) do
+    {_, end_of_current_week} = service_range_current_week(date_time)
+    beginning_of_following_week = Timex.shift(end_of_current_week, microseconds: 1)
+
+    end_of_following_week =
+      beginning_of_following_week |> Timex.end_of_week() |> end_of_service_day()
+
+    {beginning_of_following_week, end_of_following_week}
+  end
+
+  @doc """
+  Get a service range for all time afer the following week of the given date_time.
+  """
+  @spec service_range_later() :: Utils.DateTime.time_range()
+  @spec service_range_later(DateTime.t()) :: Utils.DateTime.time_range()
+  def service_range_later(date_time \\ now()) do
+    {_, end_of_following_week} = date_time |> service_range_following_week()
+    beginning_of_later = Timex.shift(end_of_following_week, microseconds: 1)
+
+    {beginning_of_later, nil}
+  end
+
+  @doc """
+  Given a time_range and a date_time, returns true if the date_time is within the time_range.
+  """
+  @spec in_range?(Utils.DateTime.time_range(), DateTime.t()) :: boolean
+  def in_range?({nil, stop}, date_time) do
+    Timex.before?(date_time, stop) || Timex.equal?(date_time, stop)
+  end
+
+  def in_range?({start, nil}, date_time) do
+    Timex.after?(date_time, start) || Timex.equal?(date_time, start)
+  end
+
+  def in_range?({start, stop}, date_time) do
+    in_range?({start, nil}, date_time) && in_range?({nil, stop}, date_time)
+  end
+
+  def in_range?(_, _), do: false
```

Comment:
This won't work as a fallback for a `{nil, nil}` range, since `def in_range?({nil, stop}, date_time)` will get matched first, and then raise an `:invalid_date` error.

Also, the way I understand these ranges is that `nil` means that there's no bound on that side of the range, so a `{nil, nil}` range would represent all time. If I'm reading that right, shouldn't `in_range?({nil, nil}, _)` return `true`?

Either way, I would recommend adding a test case for this in order to make explicit the assumption we're making here.

Reply:
I can add a guard in there, but if a user calls the function with no start or stop time in the range, they are doing something contrary to all docs. Unfortunately, the type system doesn't have a concept of at least one of them has to be a date_time.

Reply:
I bet we could make the type be something like 
```
@type time_range() :: {DateTime.t(), DateTime.t()} | {nil, DateTime.t()} | {DateTime.t(), nil}
```

It's a bit more verbose, but it does make it clear that having the range be unbounded on both ends is not expected.